src/engine.cc

```diff
@@ -0,0 +1,94 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Function> resolve;
+  Persistent<Function> reject;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
```

Comment:
I think I'd like to hold off on this for now. There's a lot of cleanups that could be landed, but I'd rather follow the current state of the code brought in from node-gtk for now, which is not using `static_cast`.